File: ruby/muse/algorithms/sorting/BubbleSortTest.rb
require "muse/algorithms/sorting/BubbleSort"
require "muse/util/SequenceBuilder"
require "muse/util/Sequences"
require "muse/util/TestRunner"

def testBubbleSort
  size = 32_768

  arr = Array.new(size)
  SequenceBuilder.packRandom(arr)

  checksum = Sequences.parityChecksum(arr)

  BubbleSort.sort(arr)

  return false if Sequences.parityChecksum(arr) != checksum

  Sequences.isSorted(arr)
end

TestRunner.pick(testBubbleSort) if __FILE__ == $PROGRAM_NAME

File: python/muse/algorithms/sorting/BubbleSortTest.py
from muse.algorithms.sorting import BubbleSort
from muse.util import SequenceBuilder, Sequences, TestRunner


def testBubbleSort() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    BubbleSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum:
        return False

    return Sequences.isSorted(arr)


def main() -> None:
    TestRunner.pick(testBubbleSort)


if __name__ == "__main__":
    main()
